src/Umbraco.Infrastructure/PropertyEditors/BlockEditorPropertyValueEditor.cs

```diff
@@ -77,6 +77,40 @@ public IEnumerable<UmbracoEntityReference> GetReferences(object? value)
         return result;
     }
 
+    /// <inheritdoc />
+    public IEnumerable<ITag> GetTags(object? value, object? dataTypeConfiguration, int? languageId)
+    {
+        var rawJson = value == null ? string.Empty : value is string str ? str : value.ToString();
+
+        BlockEditorData? blockEditorData = BlockEditorValues.DeserializeAndClean(rawJson);
+        if (blockEditorData == null)
+        {
+            return Enumerable.Empty<ITag>();
+        }
+
+        var result = new List<ITag>();
+        // loop through all content and settings data
+        foreach (BlockItemData row in blockEditorData.BlockValue.ContentData.Concat(blockEditorData.BlockValue.SettingsData))
+        {
+            foreach (KeyValuePair<string, BlockItemData.BlockPropertyValue> prop in row.PropertyValues)
+            {
+                IDataEditor? propEditor = _propertyEditors[prop.Value.PropertyType.PropertyEditorAlias];
+
+                IDataValueEditor? valueEditor = propEditor?.GetValueEditor();
+                if (valueEditor is not IDataValueTags tagsProvider)
+                {
+                    continue;
+                }
+
+                object? configuration = _dataTypeService.GetDataType(prop.Value.PropertyType.DataTypeKey)?.Configuration;
+
+                result.AddRange(tagsProvider.GetTags(prop.Value.Value, configuration, languageId));
+            }
+        }
+
+        return result;
```

Comment:
Hmm.. I don't think this would work since `GetTags` returns an `IEnumerable<ITag>`, so we'll need to iterate over the results from `tagsProvider.GetTags` and then yield return the individual tags, but I'm not entirely sure if that's a better aproach. Thoughts?

Reply:
Ah you're right, no lets keep it as is then 👍